src/hooks/useIsLogin.ts

```diff
@@ -4,11 +4,11 @@ import { type SocialSource } from '@/constants/enum.js';
 import { SORTED_SOCIAL_SOURCES } from '@/constants/index.js';
 import { useCurrentProfileAll } from '@/hooks/useCurrentProfile.js';
 
-export function useIsLogin(source?: SocialSource) {
+export function useIsLogin(source?: SocialSource | SocialSource[], strategy: 'some' | 'every' = 'some') {
     const currentProfileAll = useCurrentProfileAll();
 
     return useMemo(() => {
-        if (source) return !!currentProfileAll[source]?.profileId;
-        return SORTED_SOCIAL_SOURCES.some((x) => !!currentProfileAll[x]?.profileId);
-    }, [source, currentProfileAll]);
+        const sources = !source ? SORTED_SOCIAL_SOURCES : Array.isArray(source) ? source : [source];
+        return sources[strategy]((x) => !!currentProfileAll[x]?.profileId);
```

Comment:
`[].every(v => false) === false`, `every()` on empty array always return `true`

```suggestion
        if (!sources.length) return false
        return sources[strategy]((x) => !!currentProfileAll[x]?.profileId);
```